sklearn/utils/estimator_checks.py

```diff
@@ -737,6 +737,8 @@ def check_class_weight_classifiers(name, Classifier):
             classifier = Classifier(class_weight=class_weight)
         if hasattr(classifier, "n_iter"):
             classifier.set_params(n_iter=100)
+        if hasattr(classifier, "min_weight_fraction_leaf"):
+            classifier.set_params(min_weight_fraction_leaf=0.01)
```

Comment:
![download](https://cloud.githubusercontent.com/assets/5210848/5615282/c1881130-94ae-11e4-89f7-40bd7a95a90f.png)

It's an extremely noisy dataset that the trees seem to have a hard time fitting as the test, I believe, is written for linear models. That was to force it to learn a very small model. I could also get it to pass by forcing a decision stump here with `max_depth=1` for instance.